lua/direnv-nvim.lua

```diff
@@ -155,4 +160,42 @@ M.setup = function(opts)
 	end
 end
 
+local function _has_filetype(filetypes, ft)
+	if not type(filetypes) == "table" then
+		return false
+	end
+	for _, v in pairs(filetypes) do
+		if v == ft then
+			return true
+		end
+	end
+	return false
+end
+
+M.on_direnv_finished = function(opts, cb)
+	local pattern = { "DirenvReady", "DirenvNotFound" }
+	if opts["pattern"] then
+		pattern = opts["pattern"]
+	end
+
+	local once = false
+	if opts["once"] then
+		once = opts["once"]
+	end
+
+	vim.api.nvim_create_autocmd("User", {
+		pattern = pattern,
+		group = "direnv-nvim",
+		once = once,
+		callback = function()
+			if
```

Comment:
Maybe it could be:

```
if
    opts["filetype"] == nil
    or (opts["filetype"] == vim.bo.filetype or _has_filetype(opts["filetype"], vim.bo.filetype))
then
    cb()
end
```

That way if it is called without a filetype specified, the callback gets executed regardless?